src/aggregator/aggregator/follower_flush_mgr.go

```diff
@@ -279,12 +307,39 @@ func (mgr *followerFlushManager) CanLead() bool {
 	return true
 }
 
+// canLeadNotFlushed determines whether the follower can takeover leadership
+// for the window that was not (yet) flushed by leader.
+// This is case is possible when leader encounters a metric at some resolution
+// window for the first time in the shard it owns.
+func (mgr *followerFlushManager) canLeadNotFushed(windowSize time.Duration) bool {
+	earliestPossibleWindowStart := mgr.nowFn().Add(-windowSize)
```

Comment:
I think we might want `Truncate` here instead, to put this at the actual window start. Also, don't we still need the same logic for `openedAt` that we use in the other cases?

Reply:
Agree about `Truncate`!
Since we don't know when the never-flushed window started, we assume that it started at the most recent possible time, so our best effort is to use "now" to determine its start time.

Reply:
> Since we don't know when the never-flushed window started, we assume that it started at the most recent possible time, so our best effort is to use "now" to determine its start time.

Ahh ok I think I get what you're saying. So basically in the normal case, we find "nearest window to the open time" and compare that to the flush, whereas here we're saying "make sure you were up when the nearest window to now started". Seems right I think!